src/simGCN.c

```diff
@@ -281,17 +305,189 @@ const f32 D_80135D64 = 10000.0;
+    GXSetTevOrder(GX_TEVSTAGE14, GX_TEXCOORD_NULL, GX_TEXMAP_NULL, GX_COLOR_NULL);
+    GXSetTevOrder(GX_TEVSTAGE15, GX_TEXCOORD_NULL, GX_TEXMAP_NULL, GX_COLOR_NULL);
+
+    GXSetNumTevStages(1);
+    GXSetTevOp(GX_TEVSTAGE0, GX_REPLACE);
+    GXSetAlphaCompare(GX_ALWAYS, 0, GX_AOP_AND, GX_ALWAYS, 0);
+    GXSetZTexture(GX_ZT_DISABLE, GX_TF_Z8, 0);
+
+    for (i = 0; i < 0x10; i++) {
+        GXSetTevKColorSel(i, GX_TEV_KCSEL_1_4);
+        GXSetTevKAlphaSel(i, GX_TEV_KASEL_1);
+        GXSetTevSwapMode(i, GX_TEV_SWAP0, GX_TEV_SWAP0);
+    }
+
+    GXSetTevSwapModeTable(0, 0, 1, 2, 3);
+    GXSetTevSwapModeTable(1, 0, 0, 0, 3);
+    GXSetTevSwapModeTable(2, 1, 1, 1, 3);
+    GXSetTevSwapModeTable(3, 2, 2, 2, 3);
+
+    for (i = 0; i < 0x10; i++) {
+        GXSetTevDirect(i);
+    }
+
+    GXSetNumIndStages(0);
+    GXSetIndTexCoordScale(0, 0, 0);
+    GXSetIndTexCoordScale(1, 0, 0);
+    GXSetIndTexCoordScale(2, 0, 0);
+    GXSetIndTexCoordScale(3, 0, 0);
+
+    GXSetFog(0, 0.0f, 1.0f, 0.10000000149011612f, 1.0f, BLACK);
+    GXSetFogRangeAdj(0, 0, 0);
+    GXSetBlendMode(0, 4, 5, 0);
+    GXSetColorUpdate(1);
+    GXSetAlphaUpdate(1);
+    GXSetZMode(1, 3, 1);
+    GXSetZCompLoc(1);
+    GXSetDither(1);
+    GXSetDstAlpha(0, 0);
+    GXSetPixelFmt(0, 0);
+    GXSetFieldMask(1, 1);
+
+    GXSetCopyClear(GX_DEFAULT_BG, 0xFFFFFF);
+    GXSetCopyClamp(3);
+    GXSetDispCopyGamma(0);
+    GXSetDispCopyFrame2Field(0);
+    GXClearBoundingBox();
+    GXPokeColorUpdate(1);
+    GXPokeAlphaUpdate(1);
+    GXPokeDither(0);
+    GXPokeBlendMode(0, 0, 1, 0xF);
+    GXPokeAlphaMode(7, 0);
+    GXPokeAlphaRead(1);
+    GXPokeDstAlpha(0, 0);
+    GXPokeZMode(1, 7, 1);
+    GXSetGPMetric(0x23, 0x16);
+    GXClearGPMetric();
+
+    return 1;
+}
+#endif
 
 #pragma GLOBAL_ASM("asm/non_matchings/simGCN/simulatorUnpackTexPalette.s")
 
 #pragma GLOBAL_ASM("asm/non_matchings/simGCN/simulatorDVDShowError.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/simGCN/simulatorDVDOpen.s")
+s32 simulatorDVDOpen(char* szNameFile, DVDFileInfo* pFileInfo) {
+    s32 nStatus;
 
-#pragma GLOBAL_ASM("asm/non_matchings/simGCN/simulatorDVDRead.s")
+    while ((nStatus = DVDGetDriveStatus()) != 0) {
+        if (!simulatorDVDShowError(nStatus, 0, 0, 0)) {
+            return 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/simGCN/simulatorPlayMovie.s")
+    return DVDOpen(szNameFile, pFileInfo);
+}
+
+s32 simulatorDVDRead(DVDFileInfo* pFileInfo, void* anData, s32 nSizeRead, s32 nOffset, DVDCallback callback) {
+    s32 nStatus;
+    s32 bRetry;
+
+    if (callback == NULL) {
+        do {
+            bRetry = 0;
+            DVDReadAsyncPrio(pFileInfo, anData, nSizeRead, nOffset, NULL, 2);
+
+            while ((nStatus = DVDGetCommandBlockStatus(&pFileInfo->cb)) != 0) {
+                if (!simulatorDVDShowError(nStatus, (s32)anData, nSizeRead, nOffset)) {
+                    return 0;
+                }
+
+                if ((nStatus == 0xB) || (nStatus == -1)) {
```

Comment:
```suggestion
                if (nStatus == 11 || nStatus == -1) {
```
11 instead of 0xB because this is probably an enum

and personally I really hate that the oot codebase uses redundant parentheses so much, personally I think it makes things harder to read. There's more redundant parentheses below but I haven't really tried to enforce this so don't worry about it

Reply:
I usually put everything in parenthesis because sometimes the condition doesn't behave like I want